File: QuickStarts/VB/Stoplight/Src/ServiceInterfaces/IStoplightTimer.vb
'===============================================================================
' Microsoft patterns & practices
' Unity Application Block
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================

Imports System

Namespace StopLight.ServiceInterfaces
	Public Interface IStoplightTimer
		Property Duration() As TimeSpan
		Sub Start()
		Event Expired As EventHandler
	End Interface
End Namespace

File: quickstarts/CS/Stoplight/Src/ServiceInterfaces/IStoplightTimer.cs
// Copyright (c) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.

using System;

namespace StopLight.ServiceInterfaces
{
	public interface IStoplightTimer
	{
		TimeSpan Duration { get; set; }
		void Start();
		event EventHandler Expired;
	}
}
